### ISSUE TITLE
typechecker:handle_type_error({type_error,bin,0,...

### ISSUE THREAD

**Issue by OvermindDL1** [josefs/Gradualizer#75]
```
typechecker:handle_type_error({type_error,bin,0,           
                                                              {user_type,[{file,"test_path.erl"},{location,24}],t,[]}}) (/home/overminddl1/erlang/gradualizer/_build/default/lib/gradualizer/src/typechecker.erl, line 2807)
```
It seems binary types are not being handled by the type error path?  So should be a simple fix.  Although the fact that the binary is not passed in like it is for strings/charlists seems odd?  Also `typelib:pp_type(Ty)` does not seem to add the remote module name to the type so I just get something like `t()` instead of `test_path:t()` or so, and with how common `t` is as a type name that's not terribly helpful, plus just seeing `t` isn't really helpful either, it would be nice to get a list of types drilled down.  :-)

**Comment by OvermindDL1** [comment]
Also, this is failing because it's unable to cast a binary of an asterisk `*` to the `unicode:chardata()` type in erlang, which is just defined as `binary()`.  I'm unsure whether it's better to do strong types or soft types here, blah...

**Comment by OvermindDL1** [comment]
If I pass the `BinElements` across to the error and print out the error then I print out:
```
The binary [{bin_element,0,{string,0,"*"},default,default}] on line 0 does not have type t()
```
Line 0 is quite wrong as well, and type `t()` tells me nothing since the module this is defined in doesn't have a type `t()`, but rather `test_path.t()` is what it should be referencing.

**Comment by josefs** [comment]
Thanks for the bug report! We will want to fix this one as part of #59.
